Epic cart sprint 1

AnnaTau opened the pull request:
Стек разработки:
- SwiftUI
- MMVM
- IOS 17.6
- async/await

Выполненные задачи:
- Вёрстка ячейки коллекции
- Верстка коллекции списка покупок
- Вёрстка страницы корзины
- Вёрстка ячейки
- Вёрстка коллекции списка способов оплат
- Вёрстка страницы

Ссылка на мою канбан доску -> https://github.com/users/AnnaTau/projects/2/views/7?filterQuery=assignee%3AAnnaTau

[Декомпозиция эпика Корзина.md](https://github.com/user-attachments/files/22428558/default.md)

Затраченное и запланированное время указано в файле декомпозиции(лежит в корневой папке проекта)

Commit:
feature: добавлены модели и сервис

Commit:
feature: добавлены модели и сервис

Commit:
fix:resolve conflicts after merge

Commit:
feature: сверстана ячейка каталога

e

Commit:
feature: сверстана ячейка каталога

Commit:
feature: сверстан экран каталога и прикручен сетевой сервис

Commit:
feature: сверстан экран каталога и прикручен сетевой сервис

Commit:
feature: сверстан экран каталога и прикручен сетевой сервис

Commit:
Merge branch 'flow_catalog' into flow_catalog_new

Commit:
clean: cleaning

AnnaTau committed:
[Feature] Свёрстана ячейка для коллекции корзины

Commit:
feature: добавил сохранение состояния сортировки в UserDefaults

vaskhan committed:
Merge pull request #2 from AnnaTau/flow_catalog

Flow catalog

vaskhan committed:
Merge pull request #3 from AnnaTau/catalog_khanin

Catalog khanin

AnnaTau committed:
[Feature] Свёрстан экран корзины

AnnaTau committed:
[Clean] Добавлен project.pbxproj

AnnaTau committed:
Merge remote-tracking branch 'origin/develop' into epic_cart_sprint_1

# Conflicts:
#	iOS-FakeNFT-Extended.xcodeproj/project.pbxproj

AnnaTau committed:
[Feature] Свёрстана страница выбора оплаты, добавлена локализация и тёмная тема

AnnaTau committed:
Добавлена декомпозиция эпика Корзина

AnnaTau committed:
Добавлено затраченое время

Helvetii951 commented on iOS-FakeNFT-Extended/Resources/en.lproj/Localizable.strings:
```diff
@@ -1,13 +1,33 @@
+"by_count"       = "By count";
+"by_rating"      = "By rating";
+"by_price"       = "By price";
+"Close"          = "Close";
```
Рекомендую называть локализованные строки так:
```
"module_name.screen_name.exact_string_name" = "some string";
```

Или так:

```
"moduleName.screeName.exactStringName" = "some string";

```

Helvetii951 commented on iOS-FakeNFT-Extended/Screens/Cart/Components/CartCollectionRow.swift:
```diff
@@ -0,0 +1,92 @@
+            
+            VStack(alignment: .leading, spacing: 12) {
+                VStack(alignment: .leading, spacing: 2) {
+                    Text(item.title)
+                        .font(.appBold17)
+                    HStack(alignment: .center, spacing: 2) {
+                        ForEach(0 ..< item.rating, id: \.self) { _ in
+                            Image("YellowStar")
+                                .frame(width: 12, height: 12)
+                        }
+                        ForEach(0 ..< 5 - item.rating, id: \.self) { _ in
+                            Image("WhiteStar")
+                                .frame(width: 12, height: 12)
+                        }
+                    }
+                }
+                
+                VStack(alignment: .leading, spacing: 2){
+                    Text(String(localized: "Cart.itemPrice"))
+                        .font(.appRegular13)
+                    Text("\(String(format: "%.2f", item.price)) ETH")
+                        .font(.appBold17)
+                }
+            }
+            .padding(EdgeInsets(top: 8, leading: 0, bottom: 8, trailing: 0))
+            
+            Spacer()
+            
+            VStack {
+                Button {
+                    //TODO добавить функцию удаления из корзины
+                } label: {
+                    Image("FullBasket")
+                        .frame(width: 40, height: 40)
+                }
+            }
+            
+        }
+    }
+}
```
Рекомендую заносить константы в отдельную сущость:
```swift
private enum Constants {
        static let imageSize: CGFloat = 108
        static let cornerRadius: CGFloat = 12
        ...
    }
```

Helvetii951 commented on iOS-FakeNFT-Extended/Screens/Cart/Views/CartView.swift:
```diff
@@ -0,0 +1,101 @@
+                        VStack(alignment: .leading, spacing: 2) {
+                            Text("\(viewModel.mockedItems.count) NFT")
+                                .font(.appRegular15)
+                            Text("\(viewModel.mockedItems.map({$0.price}).reduce(0, +), specifier: "%.2f") ETH")
```
Идеально👍

Helvetii951 commented on iOS-FakeNFT-Extended/Info.plist:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>UIAppFonts</key>
+	<array>
+		<string>SF-Pro-Text-Bold.otf</string>
+		<string>SF-Pro-Text-Medium.otf</string>
+		<string>SF-Pro-Text-Regular.otf</string>
+	</array>
```
SFPro - системный шрифт в iOS. Лучше всегда пользоваться [systemFont(ofSize:weight:)](https://developer.apple.com/documentation/uikit/uifont/1619027-systemfont).

AnnaTau committed:
[Clean] Правки после ревью

Удалены шрифты из info.plist, названия в файлах локализации приведены к единому виду, размеры компонентов вью вынесены в константы
